src/Sia.Connectors.Tickets/Connector.cs

```diff
@@ -1,18 +1,43 @@
-﻿using System;
+﻿using Microsoft.Extensions.Logging;
+using Sia.Domain;
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace Sia.Connectors.Tickets
 {
-    public class Connector<TTicket>
+    public class Connector
     {
-        public Client<TTicket> Client { get; }
-        public Converter<TTicket> Converter { get; }
-
-        public Connector(Client<TTicket> client, Converter<TTicket> converter)
+        public Connector(TicketingClient client, ILoggerFactory loggerFactory)
         {
             Client = client;
-            Converter = converter;
+            Logger = loggerFactory.CreateLogger<Connector>();
+        }
+        protected TicketingClient Client { get; }
+        protected ILogger Logger { get; }
+        public virtual async Task GetData(Ticket persistedTicket)
```

Comment:
Confused about this pattern. Why not send the id and return the data back? GetData may be a confusing name. If the method is modifying the passed object without making it obvious this may be violating the "[Principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)".

Reply:
Changing the name of the method from GetData to AppendDataAsync.

We don't want the connector to be treated like a client for the associated ticketing system; it should only be used to attach data to tickets, it shouldn't be available for querying the ticketing system directly for other purposes. Deliberate inflexibility prevents misuse.

We also want to give as much control as possible to the connector; for example, a connector implementer could choose to make calls to different endpoints based on whether the ticket is primary, so that limited (or no) extra information is returned for non-primary tickets. We may eventually move to a middleware-like scheme for ticketing connectors, where the ticket gets passed through multiple layers and each connector layer can make requests based on data provided by previous layers.

Reply:
Makes sense.